src/kernels/kernel_macro.jl

```diff
@@ -0,0 +1,39 @@
+"""
+    @kernel [variance *]kernel::Kernel [l=Real/Vector / t=transform::Transform / transform::Transform]
+
+The `@kernel` macro is an helping alias to the [`transform`](@ref) function.
+The first argument should be a kernel multiplied (or not) by a scalar (variance of the kernel).
+The second argument (optional) can be a keyword :
+ - `l=ρ` where `ρ` is a positive scalar or a vector of scalar
+ - `t=transform` where `transform` is a [`Transform`](@ref) object
+One can also directly use a `Transform` object without a keyword.
+Here are some examples :
+```julia
+    k = @kernel SqExponentialKernel() l=3.0
+    k == transform(SqExponentialKernel(), ScaleTransform(3.0))
+
+    k = @kernel (MaternKernel(ν=3.0) + LinearKernel()) t=LowRankTransform(rand(4,3))
+    k == transform(KernelSum(MaternKernel(ν=3.0), LinearKernel()), LowRankTransform(rand(4,3)))
+
+    k = @kernel 4.0*ExponentiatedKernel() ScaleTransform(3.0)
+    k == ScaleTransform(transform(ExponentiatedKernel(), ScaleTransform(3.0)), 4.0)
+"""
+macro kernel(expr::Expr, arg = nothing)
+    @capture(expr, (scale_ * k_ | k_)) || throw(error("@kernel first arguments should be of the form `σ * kernel` or `kernel`"))
+    t = if @capture(arg, kw_ = val_)
+        if kw == :l
+            val
+        elseif kw == :t
+            val
+        else
+            throw(error("The additional argument could not be intepreted. Please see documentation of `@kernel`"))
+        end
+    else
+        arg
+    end
+    if isnothing(scale)
+        return esc(:(transform($k, $t)))
+    else
+        return esc(:($scale * transform($k, $t)))
+    end
```

Comment:
Maybe just (untested, so might have to be changed slightly...)
```suggestion
    t = @capture(arg, (l = val_ | t = val_)) ? val : arg
    if scale === nothing
        return :(transform($(esc(k)), $(esc(t))))
    else
        return :($(esc(scale)) * transform($(esc(k)), $(esc(t))))
    end
```
Usually it is preferable to test equality with `nothing` with `===` (even though it shouldn't make a difference here) and to escape only user-provided input in macro expressions. Otherwise one has to qualify everything else (here, e.g., `transform`) to avoid problems if a user defined a variable (or function) `transform`.

Reply:
In your code is there a way to still be able to throw an understandable error message?

Reply:
I guess something like
```suggestion
    if arg === nothing
        t = nothing
    else
        if @capture(arg, (l = val_ | t = val_))
            t = val
        else
            error("the additional argument of `@kernel` is incorrect")
        end
    end

    if scale === nothing
        return :(transform($(esc(k)), $(esc(t))))
    else
        return :($(esc(scale)) * transform($(esc(k)), $(esc(t))))
    end
```
would be possible.

Reply:
Btw `isnothing` is equivalent to `===`, in the docs of `isnothing` : 
"""
    isnothing(x)

Return `true` if `x === nothing`, and return `false` if not.
"""

Reply:
The result of the computation is the same but `===` is better for the compiler (see e.g. https://github.com/JuliaLang/julia/issues/35585#issuecomment-619053871).